lib/metadataTypes/definitions/DataExtension.definition.js

```diff
@@ -85,6 +85,7 @@ module.exports = {
     typeDescription: 'Database table schemas.',
     typeRetrieveByDefault: true,
     typeName: 'Data Extension',
+    deletionApi: 'soap',
```

Comment:
not sure how you would improve over this. 

from my POV we can _always_ overwrite the common method in case of needed customizations. but I figured it's less code if instead of introducing extra methods that just overload the super-version with 1 line (e.g. the rest methods merely passing in a simple string for the URI) it's actually much better for performance if we add 1 attribute in the definition object.

--> I was going to bring up removing all the rest of wrapper methods that don't actually have custom code in them. 

#smallerCodeBase

Reply:
given that not all classes actually support generic methods we have, I would argue it is better to keep methods approach so that we have the door open. From performance perspective what are we talking here? im not sure how it really would make much of a difference?

Reply:
see below, revert to function wrappers for now, and tackle rewrite system-wide later